Data Science Fundamentals: Python |
[Table of Contents](../index.ipynb)
- - - 
<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Exercises](./04_mod_practice.ipynb)**

# Module 1: Practice Exercises
Please remember to submit this file to Titus

### Exercise 1: Accept the user's first and last name and print them in reverse order with a space between them.

In [2]:
#insert your code here, and don't forget to comment EVERY line!
n = input("First Name:") #input command for first name
o = input("Last Name:") #input command for last name
print(o, " ", n) #print command using n and o in reverse order so last name comes first and " " to put a space between them

First Name:Anthony
Last Name:Graham
Graham   Anthony


### Exercise 2. Accept an integer (n) input from the user and compute the value of n+nn+nnn
For example if `n=4` then `n + nn + nnn = 4 + 44 + 444 = 492`

In [2]:
#insert your code here
n = int(input("Choose a whole number/integer: ")) #this code allows the user to input a whole number/integer
answer = (n + ((n*10) + n) + ((n*100) + (n*10) + n)) #this code takes the number they input, then multiplies it by 10 and adds it back to itself to get nn and then multiplies by 100 and adds that to a multiplication of n by 10 and finally also added to n by itself to get nnn
print(answer) #this prints our answer which manipulates the integer n to achieve the n + nn + nnn

Choose a whole number/integer: 4
492


### Exercise 3. Ask the user "What country are you from?" then print the following statement: "I have heard that [input] is a beautiful country!"

In [4]:
#insert your code here
country = input("What country are you from?: ") #used input to get an answer from the user about their country or origin
print("I have heard that", country, "is a beautiful country!") #inserted country variable between two strings to achieve the end sentence.

What country are you from?: America
I have heard that America is a beautiful country!


### Exercise 4. What is the output of the following python code?

In [7]:
x = 10
y = 50
if (x ** 2 > 100 and y < 100):
    print(x, y)

#Print your answer here 
#the output is nothing because x ** 2 is 100 and not greater than 100 so nothing is printed due to the indentation.

### Exercise 5: What is the output of the following addition `+` operator, and why does this code chunk execute this way??

In [5]:
a = [10, 20]
b = a
b += [30, 40]
print(a)
print(b)

[10, 20, 30, 40]
[10, 20, 30, 40]


#Print your answer here
#the output is [10, 20, 30, 40] for both a and b and they are the same because 'b += [30, 40]' adds those numbers to the original 'a' value and 'b = a'  makes the two variables equal to each other resulting in them having the same output. 

### Exercise 6. What is the output of the following code, and what operator is being used here?

In [8]:
print(2%6)

2


#Print your answer here
#the output is 2 and the operator is the modulus operator which gives the integer remainder after division occurs. 

### Exercise 7: What is the output of the following code, and what arithmetic operators are used in this code?

In [9]:
print(2 * 3 ** 3 * 4)

216


#Print your answer here
#the output for this code is 216 and the operators used are multiplicative (*) and exponential (**) 

### Exercise 8: What is a text editor?

#Print your answer here
#a text editor is a program that lets you write, edit and save text/text files, such as notepad. 

### Exercise 9: What is python?

#Print your answer here
#Python is a coding/programming language similar to C or C++, and it is very popular and known as one of the easier programming languages to learn the syntax of. 

### Exercise 10: What is jupyter notebook, what type of python environment is it, and what alternatives are there to jupyter notebook?

#Print your answer here
#Jupyter notebook is an application that allows you to run and edit notebook documents through your web browser. It is a conda environment and some alternatives are 

### Please remember to submit this file to Titus!

<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Practice](./04_mod_practice.ipynb)**
<br>
[top](#)

- - -

Copyright © 2020 Qualex Consulting Services Incorporated.